pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -20,6 +21,128 @@
 from pymc_marketing.mmm.transformers import michaelis_menten
 from pymc_marketing.mmm.utils import sigmoid_saturation
 
+from pymc_marketing.mmm.components.adstock import (
+    AdstockTransformation
+)
+from pymc_marketing.mmm.components.saturation import (
+    SaturationTransformation
+)
+
+class BudgetOptimizer:
+    """
+    A class for optimizing budget allocation in a marketing mix model.
+
+    Parameters:
+    ----------
+    adstock : AdstockTransformation
+        The adstock parameter.
```

Comment:
Is it a transformation or a parameter? This might be confused with the parameter `alpha,` for example.

(same below with `saturation` description)